### ISSUE TITLE
rootParent Layer Prop

### ISSUE THREAD

**Issue by mcost45** [visgl/deck.gl#5970]
#### Target Use case
Currently, the parent of a sublayer can be accessed via the parent property, however this only goes one layer up. For multi-view uses layer filtering can be important, and it is also important to be able to filter quickly. The most efficient way to do this is not to compare id strings, but to have a map of view ids to layer ids (that should be shown in each given view), for example:
```
private filterLayers(toFilter) {
    return viewLayerFilteringMap[toFilter.viewport.id][toFilter.layer.id];
}
```
Instead of having to cover every possible sublayer id in these maps, or track back up the parent path, I think it would be more simple to include a rootParent layer prop - so any child could easily be filtered based off the root parent it was generated from, no matter how many levels down.

#### Proposed feature
Include a 'rootParent' prop for composite layers, that refers back to the initial layer.
This could be null for the actual root parent layer.

#### To Do List 
  - [ ] Add label and assign to milestone 
  - [ ] Coding
  - [ ] Doc update
  - [ ] What’s new update 
  - [ ] Test

**Comment by ibgreen** [comment]
Comments:
- CompositeLayers can render CompositeLayers so there may be multiple roots. Would this mechanism just offer one of these? Which one?
- Do you have any actual indications that filtering on strings causes perf problems, or is it more of a general assumption.

**Comment by mcost45** [comment]
I would preference having the initial layer as the sole root layer property, to reference the root of any tree of combined composite layers. The root parents would be the top-level layers in Deck's layers prop array. So a composite layer will inherit the root parent of its direct parent, unless the direct parent has a null root parent, where in that case the direct parent also is the root parent. Mainly just because it's more simple and no use cases for something more complex immediately come to my mind.

R.e. performance it is a general assumption, based on:
- Filtering occurs twice every frame (if picking is enabled)
- Both times the filter function will run numberOfLayers x numberOfViews times (please correct me if I'm wrong)
- Depending on the complexity of the application, the filter function may contain a large set of if/else conditions to work through, each running string.startsWith('someId')

The filter function could instead by replaced by a single line only requiring two array accesses, without causing any harm I can see. Maybe this would not improve performance that much, especially in less complex applications, however it is not the only thing I am considering. I think it allows for more compact and readible code, especially when more complex/large sets of views and layers come into play - e.g. a large set of if / else conditions will not be necessary.

Another benefit of this is that it would put greater focus on the top-level layer - in terms of filtering options, to implement a map like I did above, there would be no need to understand the underlying/hidden sublayers.

**Comment by Pessimistress** [comment]
@ibgreen one could argue that `layerFilter` should only ever be called on top-level layers. Callbacks such as `onClick`, `onHover` are only invoked by top-level layers because the inner implementation of composite layers might change and should not be a concern of the user.

**Comment by Pessimistress** [comment]
New behavior is available in 8.6.0-alpha:

https://github.com/visgl/deck.gl/blob/master/docs/upgrade-guide.md#upgrading-from-deckgl-v85-to-v86
